src/GameNode.js

```diff
@@ -0,0 +1,39 @@
+const listenable = require('./listenable');
+
+class GameNode {
+    constructor(color, onClick, pos, size) {
+        this.children = new Array();
+        this.color = color;
+        this.onClick = onClick;
+        this.pos = pos;
+        this.size = size;
+        this.listeners = new Set();
+    }
+
+    addChild(node) {
+        this.children.push(node);
+        // hack to trigger onupdate. should be able to selectively re-render a subtree based on update, then tell client to only re-render updated part of canvas.
+        this.children = this.children;
+    }
+
+    addListener(listener) {
+        this.listeners.add(listener);
+    }
+
+    handleClick(x, y) {
+        this.onClick(x, y);
+    }
+
+    stateChanged() {
+        for (let listener of this.listeners) {
+            listener.handleStateChange(this);
+        }
+    }
+}
+
+const gameNode = (color, onClick, pos, size) => {
```

Comment:
I don't really like this, but it's the cleanest way I can think of right now that provides a listenable GameNode